src/main/appBadge.ts

```diff
@@ -24,8 +25,21 @@ let lastIndex: null | number = -1;
 export function setBadgeCount(count: number) {
     switch (process.platform) {
         case "linux":
-            if (count === -1) count = 0;
-            app.setBadgeCount(count);
+            if (typeof count !== "number") { //sanitize
+                throw new Error("count must be a number");
+            }
+            
+            execFile ("gdbus", [
+                "emit",
+                "--session",
+                "--object-path",
+                "/",
+                "--signal",
+                "com.canonical.Unity.LauncherEntry.Update",
+                "application://vesktop.desktop",
```

Comment:
Please note that you need to use `dev.vencord.Vesktop.deskto`p if running as Flatpak.